Leak detected when running the tests with ASAN.

```
=== release test-worker-message-port-transfer-terminate ===
Path: parallel/test-worker-message-port-transfer-terminate
Error: --- stderr ---
=================================================================
==183750==ERROR: LeakSanitizer: detected memory leaks
Indirect leak of 8000 byte(s) in 1 object(s) allocated from:
    #0 0xcdfefd in operator new(unsigned long) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0xcdfefd)
    #1 0x161f10d in node::nsolid::EnvInst::EnvInst(node::Environment*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x161f10d)
    #2 0x163b62a in node::nsolid::EnvList::AddEnv(node::Environment*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x163b62a)
    #3 0xcf604f in node::CreateEnvironment(node::IsolateData*, v8::Local<v8::Context>, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, node::EnvironmentFlags::Flags, node::ThreadId, std::unique_ptr<node::InspectorParentHandle, std::default_delete<node::InspectorParentHandle> >) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0xcf604f)
    #4 0x1321e2e in node::worker::Worker::Run() (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x1321e2e)
    #5 0x132d715 in node::worker::Worker::StartThread(v8::FunctionCallbackInfo<v8::Value> const&)::$_3::__invoke(void*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x132d715)
    #6 0x7f84ec4a1608 in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x8608)
Indirect leak of 2568 byte(s) in 1 object(s) allocated from:
    #0 0xcdfefd in operator new(unsigned long) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0xcdfefd)
    #1 0x163b61b in node::nsolid::EnvList::AddEnv(node::Environment*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x163b61b)
    #2 0xcf604f in node::CreateEnvironment(node::IsolateData*, v8::Local<v8::Context>, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, node::EnvironmentFlags::Flags, node::ThreadId, std::unique_ptr<node::InspectorParentHandle, std::default_delete<node::InspectorParentHandle> >) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0xcf604f)
    #3 0x1321e2e in node::worker::Worker::Run() (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x1321e2e)
    #4 0x132d715 in node::worker::Worker::StartThread(v8::FunctionCallbackInfo<v8::Value> const&)::$_3::__invoke(void*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x132d715)
    #5 0x7f84ec4a1608 in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x8608)
Indirect leak of 512 byte(s) in 1 object(s) allocated from:
    #0 0xcdfefd in operator new(unsigned long) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0xcdfefd)
    #1 0x161e9a8 in node::nsolid::EnvInst::EnvInst(node::Environment*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x161e9a8)
    #2 0x163b62a in node::nsolid::EnvList::AddEnv(node::Environment*) (/home/runner/work/nsolid/nsolid/out/Release/nsolid+0x163b62a)
    #3 0xcf604f in node::CreateEnvironment(node::IsolateData*, v8::Local<v8::Context>, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, node::EnvironmentFlags::Flags, node::ThreadId, std::unique_ptr<node::InspectorParentHandle, std::default_delete<node::InspectorParentHandle> >) (/home/
make[1]: *** [Makefile:552: test-ci] Error 1
make: *** [Makefile:581: run-ci] Error 2
===
=== 1 tests failed
===
Failed tests:
out/Release/nsolid --test-reporter=spec --test-reporter-destination=stdout --test-reporter=./tools/github_reporter/index.js --test-reporter-destination=stdout /home/runner/work/nsolid/nsolid/test/parallel/test-worker-message-port-transfer-terminate.js

See: https://github.com/nodesource/nsolid/actions/runs/6749560862/job/18350180559

--- a/src/nsolid/nsolid_api.cc
+++ b/src/nsolid/nsolid_api.cc
@@ -989,6 +989,22 @@ void EnvList::RemoveEnv(Environment* env) {
     stor.cb(envinst_sp, stor.data.get());
   });
 
+  // Cleanup the RunCommand queues just to be sure no dangling SharedEnvInst
+  // references are left after the Environment is gone and the EnvInst instance
+  // can be deleted.
+  EnvInst::CmdQueueStor stor;
+  while (envinst_sp->eloop_cmds_q_.dequeue(stor)) {
+    stor.cb(stor.envinst_sp, stor.data);
+  }
+
+  while (envinst_sp->interrupt_cb_q_.dequeue(stor)) {
+    stor.cb(stor.envinst_sp, stor.data);
+  }
+
+  while (envinst_sp->interrupt_only_cb_q_.dequeue(stor)) {
+    stor.cb(stor.envinst_sp, stor.data);
+  }
+
   // Don't allow execution to continue in case a RunCommand() is running in
   // another thread since they might need access to Environment specific
   // resources (like the Isolate) that won't be valid after this returns.

--- a/src/nsolid/nsolid_api.cc
+++ b/src/nsolid/nsolid_api.cc
@@ -301,7 +301,10 @@ void EnvInst::SetNodeStartupTime(const char* name, uint64_t ts) {
 std::string EnvInst::GetOnBlockedBody() {
   DCHECK(utils::are_threads_equal(creation_thread(), uv_thread_self()));
   uv_metrics_t metrics;
-  auto SharedEnvInst = GetCurrent(isolate_);
+  SharedEnvInst envinst = GetCurrent(isolate_);
+  if (envinst == nullptr) {
+    return "";
+  }
 
   HandleScope scope(isolate_);
   Local<StackTrace> stack =
@@ -316,7 +319,7 @@ std::string EnvInst::GetOnBlockedBody() {
   std::string body_string = "{";
   // TODO(trevnorris): REMOVE provider_times so libuv don't need to use the
   // floating patch.
-  uv_metrics_info(SharedEnvInst->event_loop(), &metrics);
+  uv_metrics_info(envinst->event_loop(), &metrics);
   uint64_t exit_time = metrics.loop_count == 0 ?
     env()->time_origin() : provider_times().second;
 
@@ -401,12 +404,14 @@ SharedEnvInst EnvInst::GetInst(uint64_t thread_id) {
 
 
 SharedEnvInst EnvInst::GetCurrent(Isolate* isolate) {
-  return Environment::GetCurrent(isolate)->envinst_;
+  Environment* env = Environment::GetCurrent(isolate);
+  return env == nullptr ? nullptr : env->envinst_;
 }
 
 
 SharedEnvInst EnvInst::GetCurrent(Local<Context> context) {
-  return Environment::GetCurrent(context)->envinst_;
+  Environment* env = Environment::GetCurrent(context);
+  return env == nullptr ? nullptr : env->envinst_;
 }
 
 

--- a/src/nsolid/nsolid_api.cc
+++ b/src/nsolid/nsolid_api.cc
@@ -1556,8 +1556,7 @@ void EnvList::gen_ptiles_cb_(ns_timer*) {
 }
 
 
-void EnvList::promise_tracking_(const EnvInst& envinst, bool track) {
-  SharedEnvInst envinst_sp = EnvInst::GetInst(envinst.thread_id());
+void EnvList::promise_tracking_(SharedEnvInst envinst_sp, bool track) {
   Environment* env = envinst_sp->env();
   if (env->nsolid_track_promises_fn().IsEmpty() ||
       !envinst_sp->can_call_into_js()) {
@@ -1580,12 +1579,12 @@ void EnvList::promise_tracking_(const EnvInst& envinst, bool track) {
 
 
 void EnvList::enable_promise_tracking_(SharedEnvInst envinst_sp, void*) {
-  EnvList::promise_tracking_(*envinst_sp.get(), true);
+  EnvList::promise_tracking_(envinst_sp, true);
 }
 
 
 void EnvList::disable_promise_tracking_(SharedEnvInst envinst_sp, void*) {
-  EnvList::promise_tracking_(*envinst_sp.get(), false);
+  EnvList::promise_tracking_(envinst_sp, false);
 }
 
 

--- a/src/nsolid/nsolid_api.cc
+++ b/src/nsolid/nsolid_api.cc
@@ -2516,7 +2516,7 @@ static void GetOnBlockedBody(const FunctionCallbackInfo<Value>& args) {
 static void SetupArrayBufferExports(Isolate* isolate,
                                     Local<Object> target,
                                     Local<Context> context,
-                                    std::shared_ptr<EnvInst> envinst_sp) {
+                                    SharedEnvInst envinst_sp) {
   std::unique_ptr<BackingStore> bs =
     ArrayBuffer::NewBackingStore(envinst_sp->count_fields,
                                  EnvInst::kFieldCount * sizeof(double),

--- a/src/nsolid/nsolid_api.h
+++ b/src/nsolid/nsolid_api.h
@@ -594,7 +594,7 @@ class EnvList {
   static void blocked_loop_timer_cb_(nsuv::ns_timer*);
   static void gen_ptiles_cb_(nsuv::ns_timer*);
   static void raw_metrics_timer_cb_(nsuv::ns_timer*);
-  static void promise_tracking_(const EnvInst& envinst, bool track);
+  static void promise_tracking_(SharedEnvInst envinst_sp, bool track);
   static void enable_promise_tracking_(SharedEnvInst envinst_sp, void*);
   static void disable_promise_tracking_(SharedEnvInst envinst_sp, void*);
   static void update_has_metrics_stream_hooks(SharedEnvInst, bool has_metrics);

